pkg/util/stringutils/stringutils.go

```diff
@@ -18,3 +21,18 @@ func Contains(list []string, value string) bool {
 	}
 	return false
 }
+
+// returns true if a string is formatted like an Azure resource ID
+// otherwise, returns false
+func IsResourceIDFormatted(id string) bool {
+	resourceIDPattern := "/subscriptions/.*/resourceGroups/.*/providers/.*/.*/.*"
+	// the regexp is statically defined and hard-coded, we shouldn't need to check for errors
+	match, _ := regexp.MatchString(resourceIDPattern, id)
+
+	return match
+}
+
+// ResourceType takes a resourceID as a string and returns the resource type as a string
+func ResourceType(resourceID string) string {
+	return strings.Split(resourceID, "/")[7]
+}
```

Comment:
It might be a nit pick but `stringutil` sounds like utilities for general string operations (like `LastTokenByte`).
These two functions are domain-specific string operations, and it seems natural for me to be written in a different place.

Reply:
Fair enough; I removed those functions since they were only being used in one place (each) anyways. They're simple enough that it doesn't really affect readability of the code blocks where they were being called